docs/docs/developer/style.md

```diff
@@ -0,0 +1,27 @@
+# Style
+
+## Python
+
+To ensure a consistent Python code style, we use [Ruff](https://docs.astral.sh/ruff/) for both linting and formatting.
+
+### Configuration
+
+Our configuration can be found in [KPOps](https://github.com/bakdata/kpops)' top-level `pyproject.toml`.
+
+### Editor integration
+
+Below are listed existing Ruff plugins/extensions for some of the most popular python IDEs. If you cannot find your Editor of choices or you want something more custom, [`ruff-lsp`](https://github.com/astral-sh/ruff-lsp) enables Ruff to be used in any editor that supports the LSP
+
+- [VSCode](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
+- [PyCharm](https://plugins.jetbrains.com/plugin/20574-ruff)
+
+## Markdown
+
+To ensure a consistent markdown style, we use [dprint](https://dprint.dev) to check and reformat.
+
+```shell
+dprint fmt
```

Comment:
I would just advice to stick to our git hooks, i.e. `pre-commit install`

Reply:
wdyt now?

Reply:
looks good, commented small remarks :)